core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/CaseExpression.kt

```diff
@@ -4,156 +4,108 @@ import ch.ergon.dope.DopeQuery
-            it.searchExpression.toDopeQuery(manager) to it.resultExpression.toDopeQuery(manager)
-        }
-        return DopeQuery(
-            queryString = caseDopeQuery.queryString +
-                whenThenDopeQueries.joinToString(" ", " ", " ") { "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}" } +
-                END,
-            parameters = whenThenDopeQueries.fold(emptyMap()) { parameters, query ->
-                parameters + query.first.parameters + query.second.parameters
-            },
-        )
-    }
-}
+@JvmName("simpleCaseConditionWithOutGeneric")
+fun <T : ValidType> CaseExpression<T, out ValidType>.condition(conditionExpression: SearchResult<T, out ValidType>) =
+    CaseExpression(case, firstSearchResult, *additionalSearchResult, conditionExpression)
 
-class SearchedCaseClass : Resolvable {
-    override fun toDopeQuery(manager: DopeQueryManager) = DopeQuery(queryString = CASE, parameters = emptyMap())
-}
-
-class SearchedElseCaseExpression<U : ValidType>(
-    private val case: SearchedCaseClass,
-    private val firstSearchResult: SearchResult<BooleanType, out U>,
-    private vararg val additionalSearchResult: SearchResult<BooleanType, out U>,
-    private val elseCase: UnaliasedExpression<out U>,
-) : TypeExpression<U> {
-    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val caseDopeQuery = case.toDopeQuery(manager)
-        val whenThenDopeQueries = listOf(firstSearchResult, *additionalSearchResult).map {
-            it.searchExpression.toDopeQuery(manager) to it.resultExpression.toDopeQuery(manager)
-        }
-        val elseCaseDopeQuery = elseCase.toDopeQuery(manager)
-        return DopeQuery(
-            queryString = caseDopeQuery.queryString +
-                whenThenDopeQueries.joinToString(" ", " ", " ") { "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}" } +
-                "$ELSE ${elseCaseDopeQuery.queryString} " +
-                END,
-            parameters = whenThenDopeQueries.fold(emptyMap<String, Any>()) { parameters, query ->
-                parameters + query.first.parameters + query.second.parameters
-            } + elseCaseDopeQuery.parameters,
-        )
-    }
-}
+@JvmName("simpleCaseOtherwiseWithGeneric")
+fun <T : ValidType, U : ValidType> CaseExpression<T, U>.otherwise(elseCase: TypeExpression<U>) =
+    ElseCaseExpression(case, firstSearchResult, *additionalSearchResult, elseCase = elseCase)
 
-fun case() = SearchedCaseClass()
+@JvmName("simpleCaseOtherwiseWithOutGeneric")
+fun <T : ValidType> CaseExpression<T, out ValidType>.otherwise(elseCase: TypeExpression<out ValidType>) =
+    ElseCaseExpression(case, firstSearchResult, *additionalSearchResult, elseCase = elseCase)
 
-fun <U : ValidType> SearchedCaseClass.whenThen(whenThenCondition: SearchResult<BooleanType, U>) =
-    SearchedCaseExpression(this, whenThenCondition)
+@JvmName("searchedCaseCondition")
```

Comment:
what do you mean with "searched"?

Reply:
a Searched case is just when the BooleanType is fixed